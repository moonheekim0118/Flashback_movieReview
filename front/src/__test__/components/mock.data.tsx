export const MyInfoProp = 
{ 
    id:'test',
  nickname:'test',
  Reviews:1,
  profilePic:""
}  

export const MovieProp =  
{ 
    id:'test', 
    title:'test', 
    image:'test', 
    director:'test', 
    pubDate:'test'
};

export const AuthorProp = { id:'test', nickname:'test'};

export const ReviewProp = 
{ 
    id:'test', 
    Movie: MovieProp, 
    User:AuthorProp, 
    rating:'GOOD', 
    shortComment:'test',
    character:'test',
    line:'test',
    scene:'test',
    freeComment:'test',
};