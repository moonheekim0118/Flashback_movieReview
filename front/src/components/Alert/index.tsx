import React from 'react';
import { useSelector } from 'react-redux';
import styled from 'styled-components';


// 알림창 
const Alert=()=>{
    const { message, showAlert } = useSelector((state)=>state.alert);
    // showAlert 상태일때만 띄워준다. 

    return(
        <>
            {showAlert&&
            <Container>
                {message}
            </Container>}
        </>
    );
}

const Container = styled.div`
    position:fixed;
    left:50%;
    bottom:10%;
    transform:translateX(-50%);

    padding: 25px 15px;
    
    background-color:#cc00cc;
    border-raidus:10px;
    color:#fff;
    font-size:1.2rem;
    box-shadow:0px 0px 11px -1px rgba(0,0,0,0.75);

    z-index:7000;
`;

export default Alert;