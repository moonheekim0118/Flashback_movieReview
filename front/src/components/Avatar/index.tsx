import React from 'react';
import styled from 'styled-components';

interface Props {
    nickname?:string;
    imgSrc?:string;
    size?:number;
}

// 유저 아바타
const Avatar=({nickname, imgSrc="", size=40}:Props)=>{

    return(
        <Container size={size}>
            {imgSrc? <Image src={imgSrc}/> : <Nickname size={size}>{nickname[0]}</Nickname>}
        </Container>
    );
}

const Container = styled.div<{ size:number }>`
    width:${(props)=>`${props.size}px`};
    height:${(props)=>`${props.size}px`};
`;

const Image = styled.img`
    width:100%;
    height:100%;
    border-radius:100%;
    oveflow:hidden;
    object-fit:fill;
`;

const Nickname=styled.div<{ size:number }>`
    display:flex;
    justify-content:center;
    align-items:center;
    flex-direction:column;
    text-align:center;
    width:100%;
    height:100%;
    border-radius:100%;
    overflow:hidden;
    font-size:${(props)=>`${props.size-20}px`};
    color:#fff;
    background-color:#e0e0d1;
`;

export default Avatar;