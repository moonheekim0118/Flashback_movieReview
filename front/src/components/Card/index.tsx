import React from 'react';
import styled from 'styled-components';

interface Props {
  onClick?: (e: any) => void;
  posterSrc: string;
  alertChild?: React.ReactNode;
  descriptionChild: React.ReactNode;
  optionChild?: React.ReactNode;
}

const Card = ({
  onClick,
  posterSrc,
  alertChild,
  descriptionChild,
  optionChild,
}: Props) => {
  return (
    <Container onClick={onClick}>
      {alertChild}
      <MoviePoster src={posterSrc} alt="movie poster" />
      <MovieDescription>{descriptionChild}</MovieDescription>
      {optionChild}
    </Container>
  );
};

const Container = styled.article`
  display: flex;
  align-items: center;
  width: 100%;
  height: 200px;
  position: relative;

  border-bottom: 1px solid #f4f4f4;

  padding: 10px 20px;
  margin-top: 20px;
  cursor: pointer;
`;

const MoviePoster = styled.img`
  width: 150px;
  height: 100%;
  object-fit: scale-down;
  margin-right: 20px;
`;

const MovieDescription = styled.div`
  width: 100%;
  height: 50%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export default Card;
