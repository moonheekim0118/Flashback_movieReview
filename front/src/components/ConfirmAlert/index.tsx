import React from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';

interface Props {
    text:string; // 알림창 메시지 
    clickYes:(e: any) => void; // 알림창에서 yes를 눌렀을 때 
    clickNo:(e: any) => void; // 알림차에서 No를 눌렀을 때 
}

// 확인 알림창
const ConfirmAlert=({text, clickYes, clickNo}:Props)=>{
    const root= document.getElementById('alert-root');
    
    return(
       ReactDOM.createPortal((
        <>
        <Overaly onClick={clickNo}/>
        <Container>
            <TextContainer>{text}</TextContainer>
            <ButtonContainer>
                <Button title="remove" id="submit" onClick={clickYes}>삭제</Button>
                <Button title="cancle" id="close" onClick={clickNo}>취소</Button>
            </ButtonContainer>
        </Container>
        </>
       ), root)
    );
}

const Overaly = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 8000;

  background-color: rgba(0, 0, 0, 0.3);
`


const Container = styled.div`
    display:flex;
    flex-direction:column;
    position:fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width:310px;
    height:150px;
    
    background-color:${props => props.theme.colors.bgColor};
    color:${props => props.theme.colors.fontColor};
    border-raidus:10px;
    border: 1px solid #e0e0d1;
    font-size:1.2rem;

    z-index:9000;
`;

const TextContainer = styled.div`
    width:100%;
    height:65%;
    padding: 30px 0;
    text-align:center;
    border-bottom: 1px solid #e0e0d1;
`;
const ButtonContainer = styled.div`
    height:35%;
    width:100%;
`;

const Button = styled.button<{ title: string }>`
    border: none;
    border-right:${(props)=>props.title==='remove' ? '1px solid #e0e0d1;' : 'none'};
    width:50%;
    background-color:inherit;
    padding: 17px 10px;
    color:${(props)=>props.title==='remove' ? 'red' : 'green'};
    font-weight:bold;
    font-size:1rem;

    cursor:pointer;

    &:hover{
        background-color:#e0e0d1;
    }
`;

export default ConfirmAlert;