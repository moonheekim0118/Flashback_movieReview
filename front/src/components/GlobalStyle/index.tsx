import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
    *,
    *::after,
    *::before{
        box-sizing:border-box;
        margin:0;
        padding:0;
        font-family: 'Nanum Myeongjo', serif;
    }
    *:focus{
        outline:none;
    }
    body{
        overflow-x:hidden;
        color:#000000;
        background-color:#FFF;
    }
`;

export default GlobalStyle;
