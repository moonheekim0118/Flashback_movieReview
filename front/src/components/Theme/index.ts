export const lightTheme= {
   colors:{
       bgColor:'#fff',
       fontColor:'black'
   }
}


export const darkTheme= {
  colors:{
      bgColor:'#1a1a1a',
      fontColor:'#fff'
  }
}