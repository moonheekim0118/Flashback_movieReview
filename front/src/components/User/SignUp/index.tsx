import React , { 
       useEffect, 
       useCallback , 
       useState 
} from 'react';
import { useDispatch , useSelector } from 'react-redux';
import { SIGNUP_REQUEST } from '../../../actions/user';
import { OPEN_ALERT } from '../../../actions/alert';
import Button from '../../../atoms/Buttons';
import useInput from '../../../hooks/useInput';
import useValidation from '../../../hooks/useValidation';
import {
Container,
InputContainer,
TextInput,
PasswordInput,
Label,
ErrorMessage
} from '../Login';

const SignUp=()=>{
    const dispatch = useDispatch();
    const signUpError = useSelector((state)=>state.user.signUpError);
   
    const [email, setEmail]=useInput("");
   
    const [nickname,  // 닉네임 
           setNickname, 
           nicknameLengthError
          ]=useValidation("",2,6);

    const [password, // 비밀번호  
           setPassword, 
           passwordLengthError
          ]=useValidation("",6,15);

    const [checkpassword,setCheckpassword]=useState<string>(""); // 확인용 비밀번호  

    const [passwordError, setPasswordError]=useState<boolean>(false); // 비밀번호 불일치 에러  

    // 회원가입 에러 alert
    useEffect(()=>{ 
         if(signUpError){
              dispatch({type:OPEN_ALERT,data:signUpError});
         }
    },[signUpError]);

    // submit 
    const onSubmit = useCallback((e)=>{
         e.preventDefault();
         dispatch({
              type:SIGNUP_REQUEST,
              data: {email, nickname, password}
         });
         
    },[email,nickname,password,checkpassword]);

    // 비밀번호 변경 
    const onChangeCheckPassword = useCallback((e)=>{
         setCheckpassword(e.target.value); 
         setPasswordError(e.target.value!==password); // 비밀번호가 불일치 할 경우 
    },[password]);

    return(
       <Container>
            <InputContainer>
                <Label htmlFor="user-nickname">닉네임</Label>
                <TextInput
                name="user-nickname"
                placeholder="닉네임"
                value={nickname}
                onChange={setNickname}
                />     
                {nicknameLengthError && <ErrorMessage>닉네임은 2글자 이상 6글자 이하여야 합니다.</ErrorMessage>}
           </InputContainer>  
           <InputContainer>
                <Label htmlFor="user-email">이메일</Label>
                <TextInput
                name="user-email"
                placeholder="example@example.com"
                value={email}
                onChange={setEmail}
                />     
           </InputContainer>  
           <InputContainer>
                <Label htmlFor="user-password">비밀번호</Label>
                <PasswordInput
                name="user-password"
                placeholder="6자 이상 15자 이하"
                value={password}
                onChange={setPassword}
                />    
               {passwordLengthError && <ErrorMessage>비밀번호는 6글자 이상 15글자 이하여야 합니다.</ErrorMessage>}
           </InputContainer>  
           <InputContainer>
                <Label htmlFor="user-checkpassword">비밀번호 확인</Label>
                <PasswordInput
                name="user-checkpassword"
                placeholder="비밀번호 확인"
                value={checkpassword}
                onChange={onChangeCheckPassword}
                />     
                {passwordError && <ErrorMessage>비밀번호가 일치하지 않습니다.</ErrorMessage>}
           </InputContainer> 
           <br/>
           <Button
            fill={true}
            shadow={true}
            disabled={nicknameLengthError || passwordLengthError || passwordError }
            color="purple" 
            onClick={onSubmit}
            title="회원가입"
           />
       </Container>
    );
}

export default SignUp;