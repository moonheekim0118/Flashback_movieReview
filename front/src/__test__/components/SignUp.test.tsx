import 'jsdom-global/register';
import React from 'react';
import { useSelector } from 'react-redux';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { lightTheme } from '../../components/Theme';
import SignUp from '../../components/User/SignUpForm';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(() => jest.fn()),
}));

describe('<SignUp/>', () => {
  let container;
  beforeEach(() => {
    const state = { signUpError: false };
    useSelector.mockImplementation(() => state);
    container = mount(
      <ThemeProvider theme={lightTheme}>
        <SignUp />
      </ThemeProvider>
    );
  });

  it('should match correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
