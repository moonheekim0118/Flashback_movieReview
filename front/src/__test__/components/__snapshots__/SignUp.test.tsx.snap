// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> should match correctly 1`] = `"<form class=\\"Login__Container-sc-1h2zokw-0 hoTOa-D\\"><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"user-nickname\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">닉네임</label><input name=\\"user-nickname\\" placeholder=\\"닉네임\\" type=\\"text\\" class=\\"Login__TextInput-sc-1h2zokw-2 kGKXOg\\" value=\\"\\"></div><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"user-email\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">이메일</label><input name=\\"user-email\\" placeholder=\\"example@example.com\\" type=\\"text\\" class=\\"Login__TextInput-sc-1h2zokw-2 kGKXOg\\" value=\\"\\"></div><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"user-password\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">비밀번호</label><input name=\\"user-password\\" placeholder=\\"6자 이상 15자 이하\\" type=\\"password\\" class=\\"Login__PasswordInput-sc-1h2zokw-3 hDIsIn\\" value=\\"\\"></div><div class=\\"Login__InputContainer-sc-1h2zokw-1 jPKskG\\"><label for=\\"user-checkpassword\\" class=\\"Login__Label-sc-1h2zokw-4 efwlLm\\">비밀번호 확인</label><input name=\\"user-checkpassword\\" placeholder=\\"비밀번호 확인\\" type=\\"password\\" class=\\"Login__PasswordInput-sc-1h2zokw-3 hDIsIn\\" value=\\"\\"></div><br><button fill=\\"true\\" color=\\"purple\\" class=\\"Buttons__Button-dz1ezt-0 hoQPBz\\">회원가입</button></form>"`;
